azureDatabaseUtils/src/main/java/bio/terra/workspace/azureDatabaseUtils/storage/BlobStorage.java

```diff
@@ -0,0 +1,18 @@
+package bio.terra.workspace.azureDatabaseUtils.storage;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/* Allow for blob storage mocking. */
+public interface BlobStorage {
+  default void streamOutputToBlobStorage(
+      InputStream fromStream,
+      String blobName,
+      String blobContainerName,
+      String blobContainerUrlAuthenticated) {}
+
+  default void streamInputFromBlobStorage(
+      OutputStream toStream, String blobName, String blobContainerName, String authToken) {}
```

Comment:
Why provide an empty default implementation, instead of just the definition?

Reply:
we had to do this in WDS since we need to set up BlobClient to point at the right blob first, you cant just stream from the blobName directly